src/Sarif.Converters/FortifyFprConverter.cs

```diff
@@ -0,0 +1,631 @@
+                }
+
+                _reader.Read();
+            }
+
+            _results.Add(result);
+        }
+
+        private void ParseLocationFromAnalysisInfo(Result result)
+        {
+            CodeFlow codeFlow = result.CodeFlows.First();
+            int step = 0;
+
+            _reader.Read();
+            string lastSnippetId = null;
+            while (!AtEndOf(_strings.AnalysisInfo))
+            {
+                // Note: SourceLocation is an empty element (it has only attributes),
+                // so we can't call AtStartOfNonEmpty here.
+                if (AtStartOf(_strings.SourceLocation))
+                {
+                    string snippetId = _reader.GetAttribute(_strings.SnippetAttribute);
+                    PhysicalLocation physLoc = ParsePhysicalLocationFromSourceInfo();
+
+                    var acl = new AnnotatedCodeLocation
+                    {
+                        Step = step++,
+                        PhysicalLocation = physLoc
+                    };
+
+                    // Remember the id of the snippet associated with this location.
+                    // We'll use it to fill the snippet text when we read the Snippets element later on.
+                    if (!string.IsNullOrEmpty(snippetId))
+                    {
+                        _aclToSnippetIdDictionary.Add(acl, snippetId);
+                    }
+
+                    codeFlow.Locations.Add(acl);
+
+                    // Keep track of the snippet associated with the last location in the
+                    // CodeFlow; that's the snippet that we'll associate with the Result
+                    // as a whole.
+                    lastSnippetId = snippetId;
+
+                    // Step past the empty element.
+                    _reader.Read();
+                }
+                else
+                {
+                    _reader.Read();
+                }
+            }
+
+            if (codeFlow.Locations.Any())
+            {
+                result.Locations.Add(new Location
+                {
+                    // TODO: Confirm that the traces are ordered chronologically
+                    // (so that we really do want to use the last one as the
+                    // overall result location).
+                    ResultFile = codeFlow.Locations.Last().PhysicalLocation
+                });
+
+                if (!string.IsNullOrEmpty(lastSnippetId))
+                {
+                    _resultToSnippetIdDictionary.Add(result, lastSnippetId);
+                }
+            }
+        }
+
+        private PhysicalLocation ParsePhysicalLocationFromSourceInfo()
+        {
+            string path = _reader.GetAttribute(_strings.PathAttribute);
+
+            int startLine;
+            string lineAttr = _reader.GetAttribute(_strings.LineAttribute);
+            if (!int.TryParse(lineAttr, out startLine))
+            {
+                startLine = 0;
```

Comment:
Fixed. I now use this pattern:
```
            int length = 0;
            string sizeAttribute = _reader.GetAttribute(_strings.SizeAttribute);
            if (sizeAttribute != null)
            {
                int.TryParse(sizeAttribute, out length);
            }
```

---
In reply to: [94535783](https://github.com/Microsoft/sarif-sdk/pull/554#discussion_r94535783) [](ancestors = 94535783)